### DISCUSSION TITLE
GTest download klappt nicht.

### DISCUSSION THREAD

**Discussion by Conan179** [Freetz-NG/freetz-ng#1087] [Allgemein] [answered]
Hallo erst mal.

Ich musste meine build Umgebung neu installieren, weil ich mein Windows neu installiert habe.

Beim make kommt diese fehlermeldung:

`CMake Warning at CMakeLists.txt:77 (message):
  re2c 2 or later was not found; changes to src/*.in.cc will not affect your
  build.

-- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
CMake Warning (dev) at /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:1373 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  CMakeLists.txt:233 (FetchContent_Declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

Change Dir: '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'

Run Build Command(s): /home/conan/6590/tools/build/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile
gmake[2]: Entering directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
/home/conan/6590/tools/build/bin/cmake -S/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild -B/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild --check-build-system CMakeFiles/Makefile.cmake 0
/home/conan/6590/tools/build/bin/cmake -E cmake_progress_start /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/CMakeFiles /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild//CMakeFiles/progress.marks
/usr/bin/gmake  -f CMakeFiles/Makefile2 all
gmake[3]: Entering directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
/usr/bin/gmake  -f CMakeFiles/googletest-populate.dir/build.make CMakeFiles/googletest-populate.dir/depend
gmake[4]: Entering directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
cd /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild && /home/conan/6590/tools/build/bin/cmake -E cmake_depends "Unix Makefiles" /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/CMakeFiles/googletest-populate.dir/DependInfo.cmake "--color="
gmake[4]: Leaving directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
/usr/bin/gmake  -f CMakeFiles/googletest-populate.dir/build.make CMakeFiles/googletest-populate.dir/build
gmake[4]: Entering directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
[ 11%] Performing download step (download, verify and extract) for 'googletest-populate'
cd /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps && /home/conan/6590/tools/build/bin/cmake -DCMAKE_MESSAGE_LOG_LEVEL=VERBOSE -P /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src/googletest-populate-stamp/download-googletest-populate.cmake
-- verifying file...
       file='/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src/release-1.10.0.tar.gz'
-- SHA1 hash of
    /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src/release-1.10.0.tar.gz
  does not match expected value
    expected: '9c89be7df9c5e8cb0bc20b3c4b39bf7e82686770'
      actual: 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
-- File already exists but hash mismatch. Removing...
-- Downloading...
   dst='/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src/release-1.10.0.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/google/googletest/archive/release-1.10.0.tar.gz'
CMake Error at googletest-subbuild/googletest-populate-prefix/src/googletest-populate-stamp/download-googletest-populate.cmake:163 (message):
  Each download failed!

    error: downloading 'https://github.com/google/googletest/archive/release-1.10.0.tar.gz' failed
          status_code: 1
          status_string: "Unsupported protocol"
          log:
          --- LOG BEGIN ---
          Protocol "https" not supported

  closing connection #-1

          --- LOG END ---

gmake[4]: *** [CMakeFiles/googletest-populate.dir/build.make:100: googletest-populate-prefix/src/googletest-populate-stamp/googletest-populate-download] Error 1
gmake[4]: Leaving directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
gmake[3]: *** [CMakeFiles/Makefile2:87: CMakeFiles/googletest-populate.dir/all] Error 2
gmake[3]: Leaving directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'
gmake[2]: *** [Makefile:91: all] Error 2
gmake[2]: Leaving directory '/home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild'

CMake Error at /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:1918 (message):
  Build step for googletest failed: 2
Call Stack (most recent call first):
  /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
  /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
  /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:2145 (cmake_language)
  /home/conan/6590/tools/build/share/cmake-3.31/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
  CMakeLists.txt:238 (FetchContent_MakeAvailable)

-- Configuring incomplete, errors occurred!
make[1]: *** [make/host-tools/ninja-host/ninja-host.mk:23: /home/conan/6590/source/host-tools/ninja-1.12.1/.configured] Terminated
make: *** [Makefile:47: envira] Terminated`

Die Meldung habe ich zum ersten mal.

**Comment by fda77** [comment]
"Unsupported protocol" kommt immer wenn das cmake von freetz was via https herunterladen möchte, denn es hat keine ssl.
In den host-tools gibt es openssl aber das wird noch nicht benutzt - was mal jemand machen müsste.
Warum ninja nun plötzlich nicht mehr will hab ich nicht verstanden. 
Zur Not könntest du die vorcompilierten tools verwenden, da ist ninja shcon compiliert - cmake aber auch ohne ssl

**Reply by Conan179** [reply]
Ähm, okey.

**Comment by fda77** [comment]
Du weisst ja, wenn die das zu lahm ist: selbst machen

**Reply by Conan179** [reply]
Ok, ich sag wie es ist, ich hab keinen schimmer wie.

**Comment by manfred-mueller** [comment]
Ich hatte gestern das gleiche Problem mit googletest und https und habe mir so geholfen:
 wget https://github.com/google/googletest/archive/release-1.10.0.tar.gz -P ~/freetz-ng/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src
Auf deine Umgebung angepasst, müsste es dann so aussehen:
wget https://github.com/google/googletest/archive/release-1.10.0.tar.gz -P /home/conan/6590/source/host-tools/ninja-1.12.1/builddir/_deps/googletest-subbuild/googletest-populate-prefix/src

**Comment by fda77** [comment]
Man könnte auch tools/path/cmake löschen und durch einen Link auf das systemeeigene cmake ersetzten. Das sollte https können, falls es installiert ist.

**Reply by Conan179** [reply]
Symlink auf das system interne und cmake bei https://freetz-ng.github.io/freetz-ng/PREREQUISITES/ hinzufügen?

**Reply by fda77** [reply]
Nein, wenn cmake mit openssl compiliert wird geht das so
Wenn das behoben ist gibts ein aktualisiertes tools archiv

**Comment by fda77** [comment, marked as answer]
Jetzt sollten(!) alle host tools das eigen openssl3 verwenden. Dh https download mit cmake sollten(!) nun auch funktionieren.
Das precompiled Paket hab ich auch aktualisiert, darin sollte(!) auch alles die openssl-host nutzen.

Was nicht geht ist ncftp-host auf einem "neueren" Linux zu compilieren. [..] Das mag inzwischen mit aktuellen tools nicht mehr compiliert werden. Das hat aber hier mit der openssl Problematik nichts zu tun.

**Reply by Conan179** [reply]
Der Build für alle meine Boxen lief durch, bei 3 von 4 kamm am schluss die meldung:
make[1]: warning:  Clock skew detected.  Your build may be incomplete.
Beim erneuten make lief es ophne Meldung durch.

**Reply by fda77** [reply]
"Clock skew detected. Your build may be incomplete" könnte von WSL oder der VM kommen wenn die Uhrzeit nicht passt oder Sprünge macht. Das soll vor allem passieren wenn man einen hohe CPU last hat. Das sollte aber auch schon vorher+länfer auftreten

**Reply by Conan179** [reply]
Hatte ich noch nie gehabt, nutze wsl.

**Comment by aus-sua** [comment]
Besten Dank für's ein- und umarbeiten. :-))

**Comment by fda77** [comment]
Teste zuerst mal ob auch wirklich alles geht, ich bin mir da nicht so ganz sicher... vor allem cmake+wget+uboot von den host tools

**Reply by fda77** [reply]
Und wer mit "ncftp-host" experimentieren möchte: "make ncftp-host-recompile", dazu gibt es nur die [make/host-tools/ncftp-host/ncftp-host.mk](https://github.com/Freetz-NG/freetz-ng/blob/master/make/host-tools/ncftp-host/ncftp-host.mk).
Mit altem Ubuntu 20 compilierts noch, mit neuem Fedora 41 nicht mehr.
Mit Github actions gehts auch: https://github.com/Freetz-NG/freetz-ng/actions/runs/12586908457/job/35081704281
Für die Fritzbox compilieren geht auch noch, aber das ist ein andere Package
